src/psyclone/dynamo0p3.py

```diff
@@ -2282,14 +2528,18 @@ def _stub_declarations(self, parent):
         :type parent: :py:class:`psyclone.f2pygen.SubroutineGen`
 
         '''
-        from psyclone.f2pygen import DeclGen
         api_config = Config.get().api_conf("dynamo0.3")
 
-        if not self._properties:
+        if not (self._properties or self._nfaces_h_required):
             return
 
         # Declare the necessary scalars (duplicates are ignored by parent.add)
-        for nface in list(self._arg_properties.values()):
+        scalars = list(self._arg_properties.values())
+        nfaces_h = self._symbol_table.name_from_tag("nfaces_re_h")
```

Comment:
It would be better but at the minute the symbol table associated with a DynCollection for a  kernel stub is created each time a such a collection is created (line 7496 of dynamo0p3.py). Therefore there's no state saved between them. This is the subject of #719.